### PR TITLE
Expand *getinventorylist() script command - favorite field

### PR DESCRIPTION
**Pull request by Emistry**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- return an array `@inventorylist_favorite`
- indicate whether an item in the inventory are located inside favorite tab or not.

**Issues addressed:** 
Allow npc script to detect if an item are located in favorite tab. 

Sample:
```
prontera,155,181,5	script	sample	4_F_KAFRA1,{
	getinventorylist;
	mes "Favourite Items";
	for (.@i = 0; .@i < @inventorylist_count; .@i++) {
		if (@inventorylist_favorite[.@i]) 
			mes "~ "+ getitemname(@inventorylist_id[.@i]);
	}
	close;
}
```
sample

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

File: src/map/script.c [hunk 1]
```diff
@@ -14832,48 +14832,56 @@ static BUILDIN(petloot)
...
 static BUILDIN(getinventorylist)
 {
 	struct map_session_data *sd = script->rid2sd(st);
-	char card_var[SCRIPT_VARNAME_LENGTH];
+	char script_var[SCRIPT_VARNAME_LENGTH];
+	int j = 0, k = 0;
 
-	int j=0,k;
-	if(!sd) return true;
+	if (!sd)
```
Comment (hemagx) [inline src/map/script.c]: ```sd != NULL```

File: src/map/script.c [hunk 2]
```diff
@@ -14832,48 +14832,56 @@ static BUILDIN(petloot)
...
-			pc->setreg(sd,reference_uid(script->add_variable("@inventorylist_amount"), j),sd->status.inventory[i].amount);
-			if(sd->status.inventory[i].equip) {
-				pc->setreg(sd,reference_uid(script->add_variable("@inventorylist_equip"), j),pc->equippoint(sd,i));
-			} else {
-				pc->setreg(sd,reference_uid(script->add_variable("@inventorylist_equip"), j),0);
+	for (int i = 0; i < sd->status.inventorySize; i++) {
+		if (sd->status.inventory[i].nameid > 0 && sd->status.inventory[i].amount > 0) {
+			pc->setreg(sd, reference_uid(script->add_variable("@inventorylist_id"), j), sd->status.inventory[i].nameid);
+			pc->setreg(sd, reference_uid(script->add_variable("@inventorylist_amount"), j), sd->status.inventory[i].amount);
+			if (sd->status.inventory[i].equip) {
```
Comment (hemagx) [inline src/map/script.c]: ```sd->status.inventory[i].equip != 0```

File: src/map/script.c [hunk 3]
```diff
@@ -14832,48 +14832,56 @@ static BUILDIN(petloot)
...
+				pc->setreg(sd, reference_uid(script->add_variable("@inventorylist_equip"), j), pc->equippoint(sd, i));
+			}
+			else {
```
Comment (hemagx) [inline src/map/script.c]: ``` } else { ```

File: src/map/script.c (hunk 1 above)
Comment (Emistry) [reply src/map/script.c]: done

File: src/map/script.c (hunk 2 above)
Comment (Emistry) [reply src/map/script.c]: done

File: src/map/script.c (hunk 3 above)
Comment (Emistry) [reply src/map/script.c]: done

**Pull request merged**